dd-sdk-android-core/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -249,6 +272,11 @@ object Datadog {
      */
     @JvmStatic
     @JvmOverloads
+    @Deprecated(
+        "UserInfo id property is now mandatory.",
+        ReplaceWith("setUserInfo(id!!, name, email, extraInfo, sdkCore)")
+    )
+    @JvmName("setUserInfoDeprecated")
```

Comment:
Having this is fine for me, it is only for Java-first users, which is probably the minority.

Another option to avoid conflicting overload would be to not modify the signature, but log a warning in case if `null`  is received, but many users won't notice it.